From 85e432ef241db233bdf7b5554bc565a7d777a967 Mon Sep 17 00:00:00 2001
From: sebastien baudouin <sebastien.baudouin@windriver.com>
Date: Mon, 30 Mar 2015 17:03:05 +0200
Subject: [PATCH 2/4] browser: fix error reported on bad cache policy

The cache policy is not initialised at creation time. This commit initialise it
and add log in case unsupported cache policy is used.
---
 browser/cachemanager.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/browser/cachemanager.cpp b/browser/cachemanager.cpp
index aeba1f8..fe95d63 100644
--- a/browser/cachemanager.cpp
+++ b/browser/cachemanager.cpp
@@ -48,6 +48,10 @@ cachemanager::cachemanager(QObject *parent) :
         }
         qDebug() << "Reading cachePolicy from config:" << policy;
         this->setCachePolicy(policy);
+    } else {
+        conn::brw::CACHE_POLICY policy = conn::brw::CP_ONLINE_CACHE;
+        qDebug() << "Set cachePolicy" << policy;
+        this->setCachePolicy(policy);
     }
 }
 
@@ -109,7 +113,7 @@ QNetworkRequest::CacheLoadControl cachemanager::getCacheLoadControl()
         case conn::brw::CP_ONLINE_ONLY:
             return QNetworkRequest::AlwaysNetwork;
         default:
-            qDebug() << "Illegal cache policy!";
+            qDebug() << "Illegal cache policy!" << m_policy;
             return QNetworkRequest::PreferCache;
     }
 }
-- 
1.9.1

